onnxruntime/core/session/onnxruntime_c_api.cc

```diff
@@ -444,12 +444,11 @@ ORT_API_STATUS_IMPL(OrtApis::RegisterCustomOpsLibrary, _Inout_ OrtSessionOptions
 }
 
 namespace {
-template <typename Loader>
-OrtStatus* CreateSessionImpl(_In_ const OrtEnv* env, _In_ const OrtSessionOptions* options,
-                             Loader loader, _Outptr_ OrtSession** out) {
+OrtStatus* LoadAndInitializeSession(_In_ const OrtEnv* /*env*/, _In_ const OrtSessionOptions* options,
```

Comment:
I don't think you'll ever get a nullptr for ```options``` here. Hence it can be passed as const-ref. Same applies for env. 
Secondly, the callers of LoadAndInitializeSession could get a nullptr for env from the user, in which case it is the responsibility of the caller to ensure it's not null before calling LoadAndInitializeSession. Hence it's safe to accept env as const-ref here.

Reply:
No options maybe nullptr. If you look at how the call is being made to this - you ll see that when the internal session is created, if OrtSessionOptions is nullptr, we give it default internal session options for internal session creation. Right now, we pass OrtSessionOptions as it is - so it could be nullptr. Should we just create a default OrtSessionOptions and feed it to the Load...() ?

Reply:
Resolving per offline discussion.